FLEDGE.md

```diff
@@ -736,14 +741,78 @@ For detailed specification and explainers of the trusted key-value server, see a
 - [FLEDGE Key/Value Server APIs Explainer](https://github.com/WICG/turtledove/blob/master/FLEDGE_Key_Value_Server_API.md)
 - [FLEDGE Key/Value Server Trust Model Explainer](https://github.com/privacysandbox/fledge-docs/blob/main/key_value_service_trust_model.md)
 
+##### 3.1.1 Cross-Origin Trusted Server Signals
+
+If the key-value server is on a different origin than the corresponding script, additional steps are
+needed to ensure that sensitive information is not inappropriately leaked to, or is manipulated by,
+a third party.
+
+In case of bidder signals, the following changes occur:
+1. The fetch is a cross-origin CORS fetch with `Origin:` set to the buyer script's origin.
+2. The value is passed to `crossOriginTrustedSignals` parameter, not the `trustedBiddingSignals`
+   parameter, and there is one more level of nesting denoting the server's origin, e.g:
+
+```
+{
+  'https://www.kv-server.example': {
+    'keys': {
+        'key1': arbitrary_json,
+        'key2': arbitrary_json,
+        ...},
+    'perInterestGroupData': {
+        'name1': {
+            'priorityVector': {
+                'signal1': number,
+                'signal2': number,
+                ...}
+        },
+        ...
+    }
+  }
+}
+```
+
+Seller signals have additional requirements, as the `trustedScoringSignalsURL` is provided by the
+publisher, not the seller:
+1. The seller script must provide an `Ad-Auction-Allow-Trusted-Scoring-Signals-From` response header,
+   a  [structured headers list of strings](https://www.rfc-editor.org/rfc/rfc8941) describing origins
+   fetching trusted signals from which is permitted. The trusted scoring signals fetch may not begin
+   until this header is received, in order to avoid leaking bid information to a third party.
+   This means using a cross-origin trusted server for seller information may carry a peformance
+   penalty.
+2. The fetch is a cross-origin CORS fetch with `Origin:` set to the seller script's origin.
+3. The value is passed to `crossOriginTrustedSignals` parameter, not the `trustedScoringSignals`
+   parameter, and there is one more level of nesting denoting the server's origin, e.g:
+
+```
+{
+  'https://www.kv-server.example': {
+    'renderURL': {'https://cdn.com/render_url_of_bidder': arbitrary_value_from_signals},
+    'adComponentRenderURLs': {
+        'https://cdn.com/ad_component_of_a_bid': arbitrary_value_from_signals,
+        'https://cdn.com/another_ad_component_of_a_bid': arbitrary_value_from_signals,
+        ...}
+  }
+}
+```
+
+Note that older version of Chrome did not support cross-origin trusted signals. You can query
```

Comment:
question: from a seller's perspective, we won't really have access to an environment where this can be queried before the `auctionConfig` is returned... so how will we know when to leverage this new behaviour?

Reply:
I guess it would require whatever script that bootstraps you to pass that in?

I wonder if "give me an entire table of settings" alternative to queryFeatureSupport would be helpful here, though I worry it may eventually get rather large.

Reply:
Indeed, this is something that we've been floating on the Prebid.js side for precisely this reason.

Reply:
So I just landed support for queryFeatureSupport('*') which returns a dictionary with all the features that queryFeatureSupport knows about. Explain update (to the feature detection file) is going to come shortly, just in process of going through the comments here...

Reply:
... And pushed.